pkg/envoy/server.go

```diff
@@ -506,14 +506,15 @@ func createBootstrap(filePath string, name, cluster, version string, xdsSock, eg
 }
 
 func getPortNetworkPolicyRule(sel api.EndpointSelector, l7Parser policy.L7ParserType, l7Rules api.L7Rules,
-	labelsMap cache.IdentityCache, deniedIdentities map[identity.NumericIdentity]bool) *cilium.PortNetworkPolicyRule {
+	labelsMap, deniedIdentities cache.IdentityCache) *cilium.PortNetworkPolicyRule {
 	// In case the endpoint selector is a wildcard and there are no denied
 	// identities, optimize the policy by setting an empty remote policies list
 	// to match all remote policies.
 	var remotePolicies []uint64
 	if !sel.IsWildcard() || len(deniedIdentities) > 0 {
 		for id, labels := range labelsMap {
-			if !deniedIdentities[id] && sel.Matches(labels) {
+			_, ok := deniedIdentities[id]
+			if !ok && sel.Matches(labels) {
```

Comment:
I wonder if we could cache **allowed** identities instead, as we are testing here if the identity is **not denied**?

Reply:
The reason I did not want to do this is because other state contains information about what we allow for identities (at L3 and L4), specifically `MapState`. Denied identities indicates that no traffic is allowed to those identities at L3, L4, or L7 vs. allowed identities might be interpreted as allowing traffic to / from those identities at L3, L4, and L7.